api/catalog/api/controllers/elasticsearch/utils.py

```diff
@@ -148,13 +225,22 @@ def get_result_and_page_count(
     :param results: The list of filtered result Hits.
     :return: Result and page count.
     """
+    logger = parent_logger.getChild("get_result_and_page_count")
     result_count = response_obj.hits.total.value
     natural_page_count = int(result_count / page_size)
     if natural_page_count % page_size != 0:
+        logger.debug("incrementing natural_page_count")
         natural_page_count += 1
     last_allowed_page = int((5000 + page_size / 2) / page_size)
     page_count = min(natural_page_count, last_allowed_page)
     if len(results) < page_size and page_count == 0:
+        logger.debug("setting result_count to len(results)")
```

Comment:
```suggestion
        logger.debug(f"setting result_count to {len(results)}")
```

Reply:
Oh actually this was intentionally _not_ formatted, mostly because we print `result_count` below so this information might be redundant (and thus introduce an unnecessary call to `len` for whatever that's worth).

It seems silly to think so hard about this now though, I'll update this with your suggestion (but formatted with the `name=value` way)